### DISCUSSION TITLE
ros2 "The passed action type is invalid" error when trying to dock or undock

### DISCUSSION THREAD

**Discussion by mbeavers34** [iRobotEducation/create3_docs#104] [Questions]
I get an error when I try :
```
pi@ubuntu:~$ ros2 action send_goal /General_Grievous/dock irobot_create_msgs/action/Dock "{}"
The passed action type is invalid
```

```
pi@ubuntu:~$ ros2 topic list
/General_Grievous/battery_state
/General_Grievous/cmd_audio
/General_Grievous/cmd_lightring
/General_Grievous/cmd_vel
/General_Grievous/dock
```
I'm stuck. I have tried everything I can think of. Any help would be appreciated.
Michael

**Comment by alsora** [comment]
Hi, this error usually means that the `irobot_create_msgs/action/Dock` type is not available in your environment.
Have you built/installed the `irobot_create_msgs` package?

If you installed from debian packages using `sudo apt-get install ros-galactic-irobot-create-msgs` then it should be automatically available.
If you built from sources, you have to remember to call `source <MY_WS>/install/setup.sh` for the workspace where you built the messages.

**Reply by mbeavers34** [reply]
I didn't build this so I don't know for sure.
My guess was no, it hadn't been done so I did the 'sudo apt-get install ros-galactic-irobot-create-msgs'
and still get the same message.

Other ideas?

**Reply by alsora** [reply]
I see now that the message type that you are passing is wrong.
It should be `irobot_create_msgs/action/DockServo` not `irobot_create_msgs/action/Dock`.
See the action file name here https://github.com/iRobotEducation/irobot_create_msgs/blob/main/action/DockServo.action

See a full list of actuation commands here https://iroboteducation.github.io/create3_docs/examples/actuators-cli/
